### ISSUE TITLE
Changing Villager IDs causes them not to spawn

### ISSUE THREAD

**Issue by Ankokou** [GrowthcraftCE/Growthcraft-1.7#370]
I'm having some trouble with the villagers and changing their profession IDs.

Unfortunately most mods don't offer a config for this so I'm having trouble finding out what profession IDs are and are not taken by other mods but in the case of Growthcraft's beekeeper and brewer, changing the ID from the default seems to stop them from spawning.

As far as I can tell the beekeeper is being completely overwritten by something else, it never appears and it's trades do not show up in another villager's. Changing the ID to anything else it will still not appear. (Though I admit I have not tried the Minecraft 1-4 IDs.)

The brewer shares it's profession ID with fossils and archeology's villager, overwriting it's appearance and both sets of trades appear. Changing the ID to any other number frees up the archeologist but the brewer will no longer appear (again, haven't tried minecraft's default professions)

**Comment by Ankokou** [comment]
Edit:
Changing them to the vanilla IDs will cause their trades to appear but will not overwrite the appearance. Changing them to another ID I suspect is available causes them not to appear at all, no random trades in other villagers, no growthcraft villagers.

**Comment by Sunconure11** [comment]
@IceDragon200 

Also, I have connections to the F&A devs, so I will see about getting them to add a config for their villager. I wasn't even aware they had a villager.

**Comment by IceDragon200** [comment]
@Ankokou Did you try Growthcraft by itself?

Otherwise, I don't see any problems in the code that would be causing them not to spawn.
I can understand overwriting another profession causing odd behaviour though.

@Sunconure11 Well, do that, no need to report to me :P

**Comment by Sunconure11** [comment]
That will take quite a while, but I did inform one of the devs, but a fix for stuff on their end will not come out as fast as the next dev build for GCE.

**Comment by IceDragon200** [comment]
@Ankokou Okay, I think I found the problem, we need to register the villager id and it's trade handlers separate (You'd think there would be some kind of check before adding trade handlers now wouldn't you?)

So there is a new test build up, you can find them here #307
Could you give it a twirl and lemme know if your issue is resolved?

**Comment by Ankokou** [comment]
@IceDragon200 
Ah, that would explain why in my testing before I went off to get some work done if I changed the growthcraft villager IDs to an existing one that villager would get their trades but if I changed it to a nonexistent villager profession they would never spawn.

But the test build absolutely does solve the issue I was having, a quick test spamming eggs and both appear now if given unique IDs, their trades seem fine too,

Good thinking to find that issue, and thank you for your quick work. 
I'm guessing it's safe to close this since the issue is known and resolved in the dev build?
